tests/kafkatest/tests/core/snapshot_test.py

```diff
@@ -0,0 +1,227 @@
+        node = random.choice(self.kafka.nodes)
+        self.logger.debug("Scenario 2: kill-clean-start on node %s", self.kafka.who_am_i(node))
+        self.kafka.clean_node(node)
+        self.kafka.start_node(node)
+        self.validate_success()
+
+        # Scenario -- Re-init broker after cleaning up all persistent state
+        # and create a topic where the affected broker must be the leader
+        node = random.choice(self.kafka.nodes)
+        self.logger.debug("Scenario 3: kill-clean-start-verify-produce on node %s", self.kafka.who_am_i(node))
+        self.kafka.clean_node(node)
+
+        # Create a topic where the affected broker must be the leader
+        broker_topic = "test_topic_%d" % self.topics_created
+        self.topics_created += 1
+        print("Creating topic %s" % broker_topic, flush=True)
+        topic_cfg = {
+            "topic": broker_topic,
+            "replica-assignment": self.kafka.idx(node),
+            "configs": {"min.insync.replicas": 1}
+        }
+        self.kafka.create_topic(topic_cfg)
+
+        self.kafka.start_node(node)
+
+        # Produce to the newly created topic to ensure broker has caught up
+        self.producer = VerifiableProducer(self.test_context, self.num_producers, self.kafka,
+                                           broker_topic, throughput=self.producer_throughput,
+                                           message_validator=is_int)
+
+        self.consumer = ConsoleConsumer(self.test_context, self.num_consumers, self.kafka,
+                                        broker_topic, consumer_timeout_ms=30000,
+                                        message_validator=is_int)
+        self.start_producer_and_consumer()
+        self.stop_producer_and_consumer()
+        self.validate()
+
+        self.validate_success()
+
+    @cluster(num_nodes=8)
+    @parametrize(metadata_quorum=quorum.remote_kraft)
+    # TODO: enable after concurrent startup has been merged
+    #@matrix(metadata_quorum=quorum.all_kraft)
+    def test_controller(self, metadata_quorum=quorum.colocated_kraft):
+        # Scenario -- stop a controller node and create metadata changes.
+        # Restart the controller and let it catch up
+        node = self.choose_controller_node()
+        self.logger.debug("Scenario 1: kill-create_topics-start on node %s", self.kafka.who_am_i(node))
+        self.kafka.controller_quorum.stop_node(node)
+        # Now modify the cluster to create more metadata changes
+        self.topics_created += self.create_n_topics(topic_count=10)
+        self.kafka.controller_quorum.start_node(node)
+        self.validate_success()
+
+        # Scenario -- Re-init controller with a clean kafka dir
+        node = self.choose_controller_node()
+        self.logger.debug("Scenario 2: kill-clean-start on node %s", self.kafka.who_am_i(node))
+        self.kafka.controller_quorum.clean_node(node)
+        self.kafka.controller_quorum.start_node(node)
+        self.validate_success()
```

Comment:
It is not clear to me that this is enough to verify that the controller loaded a snapshot and caught to the leader.

Reply:
I agree with @jsancio . What about if we bring down all the controller nodes and then bring them back? They'd have to load snapshot in that case, right, or else the quorum would not work?

Reply:
> What about if we bring down all the controller nodes and then bring them back?

Yes. Here are the steps in a bit more details:
1. Check that all of the controllers deleted their log segment starting at base offset 0.
2. For each controller one at a time:
  a. shutdown the controller
  b. delete all of the local data
  c. start the controller
  d. wait until it joins the quorum. I am not sure we have the tools to do this today. Maybe for now we need to sleep and file a Jira to build this tooling.
3. Create a new topic, produce to it, consume from it. This shows that the quorum is available and the brokers are fetching from it.